rllib/algorithms/dreamer/dreamer_torch_policy.py

```diff
@@ -22,6 +31,156 @@
 logger = logging.getLogger(__name__)
 
 
+class DreamerTorchPolicy(TorchPolicyV2):
+    def __init__(self, observation_space, action_space, config):
+
+        config = dict(ray.rllib.algorithms.dreamer.DreamerConfig().to_dict(), **config)
+
+        TorchPolicyV2.__init__(
+            self,
+            observation_space,
+            action_space,
+            config,
+            max_seq_len=config["model"]["max_seq_len"],
+        )
+
+        # TODO: Don't require users to call this manually.
+        self._initialize_loss_from_dummy_batch()
+
+    @override(TorchPolicyV2)
+    def loss(
+        self, model: ModelV2, dist_class: ActionDistribution, train_batch: SampleBatch
+    ) -> Union[TensorType, List[TensorType]]:
+        log_gif = False
+        if "log_gif" in train_batch:
+            log_gif = True
+
+        self.stats_dict = compute_dreamer_loss(
```

Comment:
I know this is annoying work (and is not done for some other algos either), but we should move all the loss logic into here and not have these one-off loss helper functions (further below) in the policy files.
We should do the same going forward for the other algos as well. For example, our TF Policies have these loss objects that make things look more complicated than they are.